### ISSUE TITLE
VBA_Robustness of code

### ISSUE THREAD

**Issue by hiddenllyy** [hiddenllyy/Internship-with-VBA#19]
Mincong,

I've a question, if the code takes a long time to run (like mine), can this code make some mistakes?

Because, I've noticed that in the column I want to fill, there is a long term of cells are blank. But actually, in the original report, we have the data, and I'm sure it's been imported into the EXCEL. Because when I try to do it step by step (F8), it works well and the cell is filled with correct data. But when it runs by itself, it's blank.

And which cell is blank, it's always blank... So perhaps it's my problem? It's really weird for me.

**Comment by hiddenllyy** [comment]
More details : 

because I've set a mark, so the code would search the Brouillon (original report) from the last time it's filled my table and stopped. So if there is a blank, that means it's not stopped, and for the following lines to fill, the code would still search form the same place.

For exemple,
line 100 in "Data" sheet is filled, the data comes from line 200 in "Brouillon". And I put a mark there.
Then for the 101 line in "Data" sheet, code will search from 200 in "brouillon".
If there is nothing for 101, then the marc is still on the line 200.
For line 102, it'll still begin from line 200.

And the weird thing is that, after a several blank, the code can still begin to fill the table "Data". It seems that for the former cells, it just can not recognise the content. Otherwise, it will fill it, like the others.

**Comment by mincong-h** [comment]
> if the code takes a long time to run (like mine), can this code make some mistakes?

It shouldn't. The screen might "stop" updating and feels like a bug, but this is because the UI thread cannot respond. In any case, it shouldn't make mistakes.

>  a long cells in my column are blank

You use singular and plural in the same time, do you mean one cell blank or many cells blank ? When you say blank, are you sure that : 
- cell's value is not filled, but with a huge height
- or cell's value is filled, but with many "next line" symbols ?

> Because when I try to do it step by step (F8), it works well 

You said that it takes long time to run, do you keep pressing F8 for 4 minutes or you've tested first of them and you conclude that it'll will work fine for the rest ?

**Comment by mincong-h** [comment]
I've a meeting, and I'll reply you later

**Comment by hiddenllyy** [comment]
Okay, it doesn't matter, you don't need to reply me immediately, you can reply me when you're free.

> a long cells in my column are blank

Sorry my poor English! :laughing: 
I mean a long term of cells in one column. For exemple, A2 to A10 are blank. And From A11 it become normal.

>  blank

Blank means, there is nothing. no space nor different height nor next line. it's just empty.

> do it step by step

I've demand the code begins from the line where it has the problem. For exemple A2 to A10 are empty. I demand it begins from A2, and for the whole process until the A2 has been filled with correct data, I've been using the F8... It's token me some time to do it. :smile:

**Comment by mincong-h** [comment]
It's really weird. Can you show the code ?

**Comment by hiddenllyy** [comment]
Yes, if it's not too long for you. Or I'll do some simplicities for you to read. Wait a couple of minutes.

**Comment by hiddenllyy** [comment]
Wait Mincong, I'll contact you later. I've a point de Stage with my tutor.

**Comment by hiddenllyy** [comment]
Hi Mincong,

I've solved this problem. I've tested the code and check each part with blank, and I found that all the part with blank are the files with more than 10 or even 20 zones inside. And each time the blank will stop at 10 or 20 zone, so I suppose the problem is that `zone 1` and `zone 10` have the same part `zone 1`. So after zone 10, the code begins to work from zone 11 till the end.

I didn't figure out exactly how the code can result this problem, because my code can stop searching once it gets the result. But still, I've avoid this possibility by adding a `:` at the end of each zone. So `zone 1:` and `zone 10:` are different, and it works well.

So I'll continue to move on, if I have time, I'll return to this issue and check the exact reason of that. And tell you what I've found upon. Thank you and have a good day! :laughing:
